service/container.go

```diff
@@ -212,6 +215,16 @@ func (c *container) Stop() {
 	}
 }
 
+// List all service names.
+func (c *container) List() []string {
+	names := make([]string, 0)
+	for _, e := range c.services {
+		names = append(names, e.name)
+	}
```

Comment:
To apply small buffer optimization and avoid multiply buffer grows, you can rewrite this statement in a better optimized way:
```names := make([]string, len(c.services)) <-- pre-allocate space for names
	for _, e := range c.services {
		names = append(names, e.name)
	}```

Reply:
I assume I should also set the length to 0, cos it will pre-fill it with empty strings.

Reply:
Yeah, sure. It was just pseudo-code (ctrl+c -- ctrl+v  😄). You are correct, you should set `cap` with `len` is eq to 0. Otherwite the first append would be at index `len(c.services) + 1`